collect_app/src/main/java/org/odk/collect/android/provider/ProjectsProviderAPI.kt

```diff
@@ -0,0 +1,11 @@
+package org.odk.collect.android.provider
+
+import android.net.Uri
+
+object ProjectsProviderAPI {
+    const val AUTHORITY = "org.odk.collect.android.provider.odk.projects"
```

Comment:
This naming convention seems strange to me. I would expect a single authority of `org.odk.collect.android.provider` shared between all providers -- see https://developer.android.com/guide/topics/providers/content-provider-creating. 

This does match `org.odk.collect.android.provider.odk.forms` and `org.odk.collect.android.provider.odk.instances`, though. I'm ok with matching the existing convention but let's make sure that's an explicit decision you both like, @grzesiek2010, @seadowg. I have a slight preference for going with `org.odk.collect.android.provider` in the hopes that we could one day correct the errors of the past but it's not strong.

Reply:
Yeah, I think agree after reading those docs that we should just have one authority for the app. It seems that we want to think about Collect as a "content provider" and that the `ContentProvider` implementations come together to form that. I think here it would be a good idea to change that everywhere and just use `org.odk.collect.android.provider`. We could still support `.forms` and `.instances` by keeping them in the relevant `ContentProvider` implementation's uri matcher. What do you folks think?

Reply:
Actually scratch all that. Looking at the `<provider>` documentation it does seem like we want different authorities for different kinds of data ("A list of one or more URI authorities that identify data offered by the content provider" and "Typically, it's the name of the ContentProvider subclass that implements the provider"). This seems to make even more sense to me if we end up with different permissions.

Reply:
First of all, changing anything in our current pattern would break all the apps that use this option. Is this something we would accept in the name of introducing such small adjustments?

Reply:
> We could still support .forms and .instances by keeping them in the relevant ContentProvider implementation's uri matcher.

Is that confirmed possible? If so, that might be a nice way to transition folks as needed.

To be clear, I'm not suggesting we change the other providers' authorities unless we can have the kind of migration path described above. I'm wondering about the convention we would use in an ideal world. To me it feels like we're exposing different tables of connected data. From the perspective of an integrating app, it would seem more palatable to have a single authority with different paths. The Android docs are contradictory on this point so I think we should do what feels right.

> if we end up with different permissions.

Do you see a reason for this? Seems projects, forms and instances are tightly connected and should be accessible with the same permission.

Reply:
> Is that confirmed possible? If so, that might be a nice way to transition folks as needed. It definietly works wit `UriMatcher` but then when I started thinking about the `<provider>` tag I realized that I was confused as to how Android would know which `ContentProvider` implementation to target if they all shared one authority. it turns out that this doesn't work: https://stackoverflow.com/questions/11975215/how-to-address-multiple-content-providers. Given that, the authorities we're using do make sense I think. 

@grzesiek2010 I think a change we should make given that we know we **need** a different authority for a provider is to change the URI we use (just for projects at the moment) to make it clear that projects are at the "root" of the provider. By that I mean the `CONTENT_URI` should probably be `"content://$AUTHORITY/"` instead of `"content://$AUTHORITY/projects"`. Can we make change here before this goes in?

Reply:
Yes, I agree it was strange to me too that we end up with uris with doubled forms/instances/projects at the end but just wanted to follow the same pattern, but ok I can change it for projects.

Reply:
Done!

Reply:
&*$#@? Why don't the Android docs just include the one crucial piece of information instead of making us run around?!

See also https://stackoverflow.com/questions/11975215/how-to-address-multiple-content-providers#comment15967300_11975215. I think a more sensible organization for the data we're providing would be a single provider/authority with different paths.